tests/providers/observer.js

```diff
@@ -335,13 +335,16 @@ test('validates and resets nested observers', async () => {
     { localVue: Vue, sync: false }
   );
 
+  await flush();
```

Comment:
Can you explain the purpose of the double `flush` calls?

Reply:
They are sadly necessary to make the test green in these cases. I managed to remove one but the other double calls still look necessary.

Reply:
Any idea what made them necessary @cexbrayat? This also made my tests pass with a custom mode that debounces the validation. Normal modes work without flushing twice.

Reply:
No it's hard to say. The code was previously calling `Promise.resolve` so it was always necessary to have two flush, but one was done (and hidden) by the promise waiting for 50ms. 
As of _why_ there are two flush needed, I don't know. It might be different with Vue 3/Vue Test Utils 2 though, we'll see.